frontend/src/metabase/dashboard/hoc/DashboardControls.jsx

```diff
@@ -0,0 +1,204 @@
+                delete options.night; // DEPRECATED: options.night
+
+                let hash = stringifyHashOptions(options);
+                hash = hash ? "#" + hash : "";
+
+                const { location, replace } = this.props;
+                if (hash !== location.hash) {
+                    replace({
+                        pathname: location.pathname,
+                        earch: location.search,
+                        hash
+                    });
+                }
+            };
+
+            setRefreshPeriod = refreshPeriod => {
+                this._clearRefreshInterval();
+                if (refreshPeriod != null) {
+                    this._interval = setInterval(
+                        this._tickRefreshClock,
+                        TICK_PERIOD * 1000
+                    );
+                    this.setState({ refreshPeriod, refreshElapsed: 0 });
+                    MetabaseAnalytics.trackEvent(
+                        "Dashboard",
+                        "Set Refresh",
+                        refreshPeriod
+                    );
+                } else {
+                    this.setState({
+                        refreshPeriod: null,
+                        refreshElapsed: null
+                    });
+                }
+            };
+
+            setNightMode = isNightMode => {
+                this.setState({ isNightMode });
+            };
+
+            setFullscreen = (isFullscreen, browserFullscreen = true) => {
+                if (isFullscreen !== this.state.isFullscreen) {
+                    if (screenfull.enabled && browserFullscreen) {
+                        if (isFullscreen) {
+                            screenfull.request();
+                        } else {
+                            screenfull.exit();
+                        }
+                    }
+                    this.setState({ isFullscreen });
+                }
+            };
+
+            _tickRefreshClock = async () => {
+                let refreshElapsed = (this.state.refreshElapsed || 0) +
+                    TICK_PERIOD;
+                if (this.state.refreshPeriod && refreshElapsed >= this.state.refreshPeriod) {
+                    this.setState({ refreshElapsed: 0 });
+                    await this.props.fetchDashboard(
+                        this.props.dashboardId,
+                        this.props.location.query
+                    );
+                    this.props.fetchDashboardCardData({
+                        reload: true,
+                        clear: false
+                    });
+                } else {
+                    this.setState({ refreshElapsed });
+                }
+            };
+
+            _clearRefreshInterval() {
+                if (this._interval != null) {
+                    clearInterval(this._interval);
+                }
+            }
+
+            _showNav(show) {
+                const nav = document.querySelector(".Nav");
```

Comment:
Again a comment regarding old code, but this would ideally be in Redux state instead of doing ad-hoc DOM mutation.